src/main/java/uk/ac/ebi/ega/egafuse/service/EgaFuse.java

```diff
@@ -65,7 +65,7 @@ public int readdir(String path, Pointer buf, FuseFillDir filter, @off_t long off
     public int getattr(String path, FileStat stat) {
         EgaPath p = getPath(path);
         if (p != null) {
-            p.getattr(stat);
+            p.getattr(stat, getContext().uid.get(), getContext().gid.get());
```

Comment:
it's weird to pass the uid and gid into a method which is supposed to just be a get. If you want always for the uid and gid to be this values, I think it would be better to just set them on the stat object here after the call.